x-pack/legacy/plugins/siem/public/containers/case/configure/use_configure.test.tsx

```diff
@@ -0,0 +1,326 @@
+      const { result, waitForNextUpdate } = renderHook<string, ReturnUseCaseConfigure>(() =>
+        useCaseConfigure(args)
+      );
+      await waitForNextUpdate();
+      await waitForNextUpdate();
+      result.current.refetchCaseConfigure();
+      expect(spyOnGetCaseConfigure).toHaveBeenCalledTimes(2);
+    });
+  });
+
+  test('set isLoading to true when fetching case configuration', async () => {
+    await act(async () => {
+      const { result, waitForNextUpdate } = renderHook<string, ReturnUseCaseConfigure>(() =>
+        useCaseConfigure(args)
+      );
+      await waitForNextUpdate();
+      await waitForNextUpdate();
+      result.current.refetchCaseConfigure();
+
+      expect(result.current.loading).toBe(true);
+    });
+  });
+
+  test('persist case configuration', async () => {
+    await act(async () => {
+      const { result, waitForNextUpdate } = renderHook<string, ReturnUseCaseConfigure>(() =>
+        useCaseConfigure(args)
+      );
+      await waitForNextUpdate();
+      await waitForNextUpdate();
+
+      result.current.persistCaseConfigure({
+        connectorId: '456',
+        connectorName: 'My Connector 2',
+        closureType: 'close-by-pushing',
+      });
+
+      expect(result.current).toEqual({
+        loading: false,
+        persistLoading: true,
+        refetchCaseConfigure: result.current.refetchCaseConfigure,
+        persistCaseConfigure: result.current.persistCaseConfigure,
+      });
+    });
+  });
+
+  test('save case configuration - postCaseConfigure', async () => {
+    // When there is no version, a configuration is created. Otherwise is updated.
+    const spyOnGetCaseConfigure = jest.spyOn(api, 'getCaseConfigure');
+    spyOnGetCaseConfigure.mockImplementation(() =>
+      Promise.resolve({
+        ...caseConfigurationCamelCaseResponseMock,
+        version: '',
+      })
+    );
+
+    const spyOnPostCaseConfigure = jest.spyOn(api, 'postCaseConfigure');
+    spyOnPostCaseConfigure.mockImplementation(() =>
+      Promise.resolve({
+        ...caseConfigurationCamelCaseResponseMock,
+        connectorId: '456',
+        closureType: 'close-by-pushing',
+        connectorName: 'My Connector 2',
+      })
+    );
+
+    await act(async () => {
+      const { result, waitForNextUpdate } = renderHook<string, ReturnUseCaseConfigure>(() =>
+        useCaseConfigure(args)
+      );
+      await waitForNextUpdate();
+      await waitForNextUpdate();
+
+      result.current.persistCaseConfigure({
+        connectorId: '456',
+        connectorName: 'My Connector 2',
+        closureType: 'close-by-pushing',
+      });
+
+      await waitForNextUpdate();
+
+      expect(args.setConnector).toHaveBeenCalled();
+      expect(args.setConnector).toHaveBeenNthCalledWith(2, '456');
+      expect(args.setClosureType).toHaveBeenCalled();
+      expect(args.setClosureType).toHaveBeenNthCalledWith(2, 'close-by-pushing');
+      expect(args.setCurrentConfiguration).toHaveBeenCalled();
```

Comment:
redundant

Reply:
these comments are redundant, I'm going to stop leaving them. you get the idea ;)